examples/games/desktoy.rs

```diff
@@ -0,0 +1,396 @@
+    if primary_window.decorations {
+        primary_window.cursor.hit_test = true;
+        return;
+    }
+
+    // If the cursor is not within the window we don't need to update whether the window is clickable or not
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    // If the cursor is within the radius of the Bevy logo make the window clickable otherwise the window is not clickable
+    let bevy_logo_transform = q_bevy_logo.single();
+    primary_window.cursor.hit_test = bevy_logo_transform
+        .translation
+        .truncate()
+        .distance(cursor_world_pos)
+        < BEVY_LOGO_RADIUS;
+}
+
+/// Start the drag operation and record the offset we started dragging from
+fn start_drag(
+    mut commands: Commands,
+    cursor_world_pos: Res<CursorWorldPos>,
+    q_bevy_logo: Query<&Transform, With<BevyLogo>>,
+) {
+    // If the cursor is not within the primary window skip this system
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    // Get the offset from the cursor to the Bevy logo sprite
+    let bevy_logo_transform = q_bevy_logo.single();
+    let drag_offset = bevy_logo_transform.translation.truncate() - cursor_world_pos;
+
+    // If the cursor is within the Bevy logo radius start the drag operation and remember the offset of the cursor from the origin
+    if drag_offset.length() < BEVY_LOGO_RADIUS {
+        commands.insert_resource(DragOperation(drag_offset));
+    }
+}
+
+/// Stop the current drag operation
+fn end_drag(mut commands: Commands) {
+    commands.remove_resource::<DragOperation>();
+}
+
+/// Drag the Bevy logo
+fn drag(
+    drag_offset: Res<DragOperation>,
+    cursor_world_pos: Res<CursorWorldPos>,
+    time: Res<Time>,
+    mut q_bevy_logo: Query<&mut Transform, With<BevyLogo>>,
+    mut q_pupils: Query<&mut Pupil>,
+) {
+    // If the cursor is not within the primary window skip this system
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    // Get the current Bevy logo transform
+    let mut bevy_transform = q_bevy_logo.single_mut();
+
+    // Calculate the new translation of the Bevy logo based on cursor and drag offset
+    let new_translation = cursor_world_pos + drag_offset.0;
+
+    // Calculate how fast we are dragging the Bevy logo (unit/second)
+    let drag_velocity =
+        (new_translation - bevy_transform.translation.truncate()) / time.delta_seconds();
+
+    // Update the translation of Bevy logo transform to new translation
+    bevy_transform.translation = new_translation.extend(bevy_transform.translation.z);
+
+    // Add the cursor drag velocity in the opposite direction to each pupil.
+    // Remember pupils are using local coordinates to move. So when the Bevy logo moves right they need to move left to
+    // simulate inertia, otherwise they will move fixed to the parent.
+    for mut pupil in &mut q_pupils {
+        pupil.velocity -= drag_velocity;
```

Comment:
This effect seems really intense right now. I tried scaling it by `0.2` and it still seemed plenty googly.

Reply:
I think the decay should also be scaled up to something like 2.0 * delta time. Feels much more natural that way.

Reply:
I've toned down the bounciness and added velocity loss when the pupil bounces off the wall.